torch/distributed/tensor/_ops/utils.py

```diff
@@ -158,8 +158,21 @@ def prod(xs: Iterable[int]) -> int:
     return functools.reduce(operator.mul, xs, 1)
 
 
-def is_tensor_shardable(shape: Sequence[int], spec: DTensorSpec) -> bool:
+def is_tensor_shardable(
+    shape: Sequence[int],
+    spec: DTensorSpec,
+    allow_unbacked_sharding: Optional[bool] = None,
+) -> bool:
     """Check if the shape is shardable according to the spec."""
```

Comment:
update the comment explaining how None/True/False behave differently for allow_unbacked_sharding